src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private Object[] objects;
+    private int size;
+
+    public ArrayList() {
+        objects = new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        checkingSizeOfList();
+        objects[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index value. "
+                    + "Index out of bounds.");
+        }
+        checkingSizeOfList();
+        if (index != size) {
+            System.arraycopy(objects, index, objects, index + 1, size - index);
+        }
+        objects[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (validationIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index value. "
+                    + "Index out of bounds.");
+        }
+        return (T) objects[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (validationIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index value. "
+                    + "Index out of bounds.");
+        }
+        objects[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (validationIndex(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index value. "
+                    + "Index out of bounds.");
+        }
+        Object object = objects[index];
+        removeObject(index);
+        return (T) object;
     }
 
     @Override
-    public T remove(T element) {
-        return null;
+    public T remove(T value) {
+        int index = findByValue(value);
+        if (index == -1) {
+            throw new NoSuchElementException("This value is missing in list");
+        }
+        removeObject(index);
+        return value;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkingSizeOfList() {
+        if (size == objects.length) {
+            objects = Arrays.copyOf(objects,size + size / 2);
+        }
+    }
+
+    private boolean validationIndex(int index) {
+        return index >= size || index < 0;
+    }
+
+    private int findByValue(T value) {
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(objects[i], value)) {
+                return i;
+            }
+        }
+        return -1;
+    }
```

Comment:
creating this private helper method is redundant
just move this logic to place where you use it

Reply:
also you are not allowed to use `Objects.equals()`